src/vivarium_public_health/metrics/mortality.py

```diff
@@ -167,23 +149,50 @@ def calculate_ylls(self, x: pd.DataFrame) -> float:
 
     def write_mortality_results(
         self,
-        cause_state: Union[str, DiseaseState, RiskAttributableDisease],
         measure: str,
         results: pd.DataFrame,
     ) -> None:
-        measure_name = measure.split("_due_to_")[0]
-        kwargs = {
-            "entity_type": (
-                cause_state.cause_type if not isinstance(cause_state, str) else "cause"
-            ),
-            "entity": cause_state.model if not isinstance(cause_state, str) else cause_state,
-            "sub_entity": cause_state.state_id if not isinstance(cause_state, str) else None,
-            "results_dir": self.results_dir,
-            "random_seed": self.random_seed,
-            "input_draw": self.input_draw,
-        }
-        write_dataframe_to_parquet(
+        """Format dataframe and write out"""
+
+        results = results.reset_index()
+
+        if "cause_of_death" in results.columns:
+            results.rename(columns={"cause_of_death": COLUMNS.ENTITY}, inplace=True)
+        else:
+            # self.aggregate_causes is True
+            results[COLUMNS.ENTITY] = "all_causes"
+
+        results = results[results[COLUMNS.ENTITY] != "not_dead"]
+        results[COLUMNS.MEASURE] = measure
+
+        results.loc[results[COLUMNS.ENTITY] == "other_causes", COLUMNS.ENTITY_TYPE] = "cause"
```

Comment:
You could replace this line and remove analogous the all_causes line below with 
```suggestion
        results[COLUMNS.ENTITY_TYPE] = "cause"
```